enigma/src/main/java/cuchaz/enigma/analysis/index/JarIndexer.java

```diff
@@ -27,6 +27,28 @@ default void indexFieldReference(MethodDefEntry callerEntry, FieldEntry referenc
 	default void indexLambda(MethodDefEntry callerEntry, Lambda lambda, ReferenceTargetType targetType) {
 	}
 
+	default void indexInnerClass(ClassDefEntry classEntry, InnerClassData innerClassData) {
+	}
+
+	default void indexOuterClass(ClassDefEntry classEntry, OuterClassData outerClassData) {
+	}
+
 	default void processIndex(JarIndex index) {
 	}
+
+	record InnerClassData(String name, String innerName, String outerName, int access) {
+		public boolean hasInnerName() {
+			return innerName != null;
+		}
+
+		public boolean hasOuterName() {
+			return outerName != null;
+		}
+	}
+
+	record OuterClassData(String owner, String name, String descriptor) {
```

Comment:
These names are wrong names from ASM, should've been like `EnclosingMethodData`. Also `hasEnclosingMethod` is wrong too; it returns that whether the given class is enclosed in a regular method instead of `<init>` or `<clinit>` (read the italics in https://docs.oracle.com/javase/specs/jvms/se19/html/jvms-4.html#jvms-4.7.7)